### PR TITLE
Add simple polars support

### PR DESCRIPTION
**Pull request by Hoxbro**
resolves https://github.com/holoviz/hvplot/issues/1023

This adds a simple Polars support to hvplot by converting it to a Pandas DataFrame before putting it into the machinery of hvplot. I'm trying to convert only the necessary columns for the  DataFrame and LazyDataFrame. 

Using the API as described [here](https://pola-rs.github.io/polars/py-polars/html/reference/api.html).

``` python
import hvplot.polars  # noqa
import polars as pl
from bokeh.sampledata.autompg import autompg

df = pl.DataFrame(autompg)
s = df.select("mpg").to_series()
ldf = df.lazy()

type(df), type(s), type(ldf)

df.hvplot(y="cyl")
s.hvplot()
ldf.hvplot.scatter()
```

image

### CODE REVIEW

**Comment by itsgifnotjiff** [comment]
Thank you so much Mr. @Hoxbro . What do you mean by simple? Is there something that a Pandas DataFrame backed HvPlot/Holoviews object can fundementally do that a Polars one can not?

**Comment by philippjfr** [comment]
>What do you mean by simple? Is there something that a Pandas DataFrame backed HvPlot/Holoviews object can fundementally do that a Polars one can not?

This PR simply converts the data to a pandas DataFrame before passing it to the existing codepaths. The longer term and more efficient approach will be to implement a data backend for HoloViews that natively understands and works with polars dataframes, avoiding any copies that occur in the polars -> pandas conversion and potentially executing specific computations, e.g. histograms, aggregations etc., directly with polars.

**Comment by itsgifnotjiff** [comment]
Thank you Mr. @philippjfr . I have a Panel app that uses Polars to load 12 GB SQLite and cast it to a Pandas DataFrame and I always wondered how it would perform using Polars alone.

**Review by maximlt — COMMENTED** [review]
Except a few minor changes this looks good to me. This is going to make polars users very happy! I wish we had a place where to expose that more in the docs, the front-page isn't the best place.

Note to self, the release notes should mention that this is basic support and the data is casted to Pandas objects. By the way @Hoxbro `Series` don't need to be casted?

File: setup.py [hunk 1]
```diff
@@ -60,6 +60,7 @@ def get_setup_version(reponame):
     'pooch',
     'scipy',
     'ipywidgets',
+    'polars',
```
[the remark this answers is not in the document]
Comment (Hoxbro) [reply setup.py:63]: > polars didn't look difficult to install at all, it actually looks like it has no dependency at all, which helps! 

polars is written in Rust, so it has dependencies we just don't see them. 

> It's also good to think about re-packagers/distros - we occasionally see them opening issues on our repos

Agree, but I don't think there would have been a problem here as the tests are inside `try/except`.

File: hvplot/polars.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+try:
+    import polars as pl
+except:
+    raise ImportError(
+        "Could not patch plotting API onto Polars. Polars could not be imported."
+    )
+
+
+@pl.api.register_dataframe_namespace("hvplot")
```
[the remark this answers is not in the document]
Comment (Hoxbro) [reply hvplot/polars.py:15]: I thought about it. But I think it makes the code unnecessarily complex.   

Most other data backends also don't need to change the `hvPlotTabular` itself like is needed here for Polars.  Dask does it with `Interactive`, outside the `patch`. The registering of the API with a decorator in Polars just makes sense to me to do directly for the `hvPlotTabularPolars` class.

File: hvplot/tests/plotting/testcore.py [hunk 3]
```diff
@@ -1,21 +1,66 @@
...
+    df.hvplot(y=y, kind=kind)
+
+
+@series_kinds
+def test_series_pandas(kind):
+    ser = pd.Series(np.random.rand(10), name="A")
+    ser.hvplot(kind=kind)
+
+
+@pytest.mark.skipif(skip_polar, reason="polars not installed")
```
[the remark this answers is not in the document]
Comment (Hoxbro) [reply hvplot/tests/plotting/testcore.py:51]: Yeah, just wanted to test if everything basic runs with Polars, and when I was at it also did it with Dask.

File: hvplot/polars.py [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+        else:
+            raise ValueError(
+                "Only Polars DataFrame, Series, and LazyFrame are supported"
+            )
+
+        return HoloViewsConverter(data, x, y, kind=kind, **params)
+
+
+def patch(name="hvplot", extension="bokeh", logo=False):
+    import hvplot.pandas  # noqa
```
[the remark this answers is not in the document]
Comment (Hoxbro) [reply hvplot/polars.py:59]: I don't think it is required so I have removed it.

File: hvplot/polars.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (maximlt) [reply hvplot/polars.py:16]: `dask.py` creates an Interactive subclass indeed but the patch happens in the `patch` function. I will suggest a change that I think will make both of us happy.

File: hvplot/polars.py [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+        else:
+            raise ValueError(
+                "Only Polars DataFrame, Series, and LazyFrame are supported"
+            )
+
+        return HoloViewsConverter(data, x, y, kind=kind, **params)
+
+
+def patch(name="hvplot", extension="bokeh", logo=False):
+    post_patch(extension, logo)
```
Comment (maximlt) [inline hvplot/polars.py:59]: I'm suggesting to do actual patching/registering in `patch`.

```suggestion
    pl.api.register_dataframe_namespace(name)(hvPlotTabularPolars)
    pl.api.register_series_namespace(name)(hvPlotTabularPolars)
    pl.api.register_lazyframe_namespace(name)(hvPlotTabularPolars)

    post_patch(extension, logo)
```

File: hvplot/polars.py [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+    import polars as pl
+except:
+    raise ImportError(
+        "Could not patch plotting API onto Polars. Polars could not be imported."
+    )
+
+
+@pl.api.register_dataframe_namespace("hvplot")
+@pl.api.register_series_namespace("hvplot")
+@pl.api.register_lazyframe_namespace("hvplot")
```
Comment (maximlt) [inline hvplot/polars.py:18]: Goes hand in hand with the other suggestion.

```suggestion
```

File: hvplot/polars.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (Hoxbro) [reply hvplot/polars.py:16]: I would rather have simplicity over consistency, and I think adding the decorator inside the `patch` adds complexity. 

With that being said, your changes look good to me. But If we want consistency shouldn't the Polar import also be in there?

[the remark this answers is not in the document]
Comment (maximlt) [reply hvplot/polars.py:16]: Yes please!

**Pull request merged**